R/kartverket.R

```diff
@@ -0,0 +1,211 @@
+#' Get address info (safe method)
+#' @inheritParams get_address_info
+#' @noRd
+get_address_info_safe <-
+  purrr::possibly(get_address_info_single, otherwise = NULL, quiet = FALSE)
+
+#' Find addresses within a geographical area
+#'
+#' Search for Norwegian addresses within a certain radius from a geographical
+#' point.
+#'
+#' @details Advanced users might want to look into the additional possible
+#' parameters. For a list of all accepted query parameters see
+#' [create_params_kv_punktsok()] or refer to the [API
+#' documentation](https://ws.geonorge.no/adresser/v1/#/default/get_sok) for
+#' further details (in Norwegian only).
+#'
+#' The function returns a data.frame or list by default. If you set
+#' `raw_response` to `TRUE`, the raw response from the API will be returned
+#' instead. Note that the response will then be returned silently.
+#'
+#' @param x A vector or list with latitude and longitude coordinates.
+#' @param radius An integer value with the radius for the query, in whole
+#'   meters.
+#' @param crs An integer or vector with the input and output coordinate
+#'   reference system. Default value is ETRS 89 for latitude and longitude data
+#'   (EPSG:4258).
+#' @inheritParams get_address_info
+#' @inheritParams get_entity
+#'
+#' @return data.frame or list
+#' @export
+#' @examples
+#' # Using a vector
+#' df <- find_address_from_point(c(lat = 63.42805, lon = 10.39679))
+#'
+#' # Using a list
+#' dl <- find_address_from_point(
+#'   list(c(lat = 59.91364, lon = 10.7508),
+#'        c(lat = 63.42805, lon = 10.39679)))
+#'
+#' # Using a data.frame
+#' dl <- find_address_from_point(
+#'     data.frame(lat = c(59.91364, 63.42805),
+#'                lon = c(10.7508, 10.39679)))
+#'
+find_address_from_point <- function(x, radius = 50, crs = 4258, ...,
+                                    raw_response = FALSE) {
+  UseMethod('find_address_from_point')
+}
+
+#' find_address_from_point (numeric)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.numeric <- function(x, radius = 50, crs = 4258, ...,
+                                            raw_response = FALSE) {
+  res <- find_address_from_point_safe(x, radius = radius, crs = crs,
+                                      raw_response = raw_response)
+  if (raw_response) return(invisible(res))
+  res
+}
+
+#' find_address_from_point (list)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.list <- function(x, radius = 50, crs = 4258, ...,
+                                         raw_response = FALSE) {
+  dl <- lapply(x, find_address_from_point.numeric, radius = radius, crs = crs,
+               ..., raw_response = raw_response)
+  if (raw_response) return(invisible(dl))
+  if (length(dl) == 1) dl <- dl[[1]]
+  dl <- dl[lengths(dl) != 0]
+  dl
+}
+
+#' find_address_from_point (data.frame)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.data.frame <- function(x, radius = 50, crs = 4258, ...){
```

Comment:
Consider adding argument to select only the closest address when multiple addresses are returned. This can be useful if you only need reverse geocoding for a data.frame (but with a caveat that flats will have multiple addresses within the same geo point).

Reply:
Yeah. That should work great. Default setting should then be `FALSE` (at least for numeric and list methods).

Reply:
This has been added, defaulting to FALSE for all methods.